viewing_party/party.py

```diff
@@ -1,23 +1,151 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title == None or genre == None or rating == None:
+        movie_data = None
+
+    else:
+        movie_data = {
+            'title': title,
+            'genre': genre,
+            'rating': rating
+            }
+
+    return movie_data
+
+
+def add_to_watched(user_data, movie):
+    user_data['watched'].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data['watchlist'].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, MOVIE_TITLE_1):
+    for movies in user_data.values():
+        for movie in movies:
+            if movie["title"] == MOVIE_TITLE_1:
+                add_movie = movies.pop(movies.index(movie))
+                user_data["watched"].append(add_movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    try:
+        total = 0
+        num_movies = 0
+
+        for movies in user_data.values():
+            for movie in movies:
+                num_movies += 1
+                total += movie['rating']
+        average = total / num_movies
+    except ZeroDivisionError:
+        average = 0.0
+
+    return average
+
+
+def get_most_watched_genre(user_data):
+    if user_data["watched"]:
+        most_popular_genre = []
+        for movies in user_data.values():
+            for movie in movies:
+                most_popular_genre.append(movie['genre'])
+        user_data = max(set(most_popular_genre), key = most_popular_genre.count)
+    else:
+        user_data = None
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    unique_watched = []
+    friend_1_watched_movies = user_data["friends"][0]["watched"]
+    friend_2_watched_movies = user_data["friends"][1]["watched"]

```

Comment:
This works great if we know we will only ever have 2 friends lists to compare to, but say we have 3 friends? Or 10? We want out code to be able to be flexible enough that it can scale to larger inputs without us having to make changes to our code. 

Like all coding problems, there's several approaches we could take! One way could be to use nested loops to make sure we touch all the data we need to. Another approach would be to make a helper function that flattens all the movies in the various friend's lists into a single list with no duplicates, then loop over it to check for unique watched movies the same way you are doing below.